devicetypes/smartthings/zwave-rgbw-bulb.src/zwave-rgbw-bulb.groovy

```diff
@@ -0,0 +1,324 @@
+/**
+ *  Copyright 2020 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *  Z-Wave RGBW Bulb, adapted from Aeon LED Bulb 6 Multi-Color
+ *
+ *  Author: SmartThings
+ *  Date: 2020-01-27
+ */
+
+metadata {
+	definition (name: "Z-Wave RGBW Bulb", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "oic.d.light", mnmn: "SmartThings", vid: "generic-rgbw-color-bulb") {
+		capability "Switch Level"
+		capability "Color Control"
+		capability "Color Temperature"
+		capability "Switch"
+		capability "Refresh"
+		capability "Actuator"
+		capability "Sensor"
+		capability "Health Check"
+		capability "Configuration"
+
+		/*
+		 * Relevant device types:
+		 *
+		 * * 0x11 GENERIC_TYPE_SWITCH_MULTILEVEL
+		 * * 0x01 SPECIFIC_TYPE_POWER_SWITCH_MULTILEVEL
+		 * * 0x02 SPECIFIC_TYPE_COLOR_TUNABLE_MULTILEVEL
+		 *
+		 * Plausible command classes we might see in a color light bulb:
+		 *
+		 * 0x98 COMMAND_CLASS_SECURITY
+		 * 0x5E COMMAND_CLASS_ZWAVEPLUS_INFO_V2
+		 * 0x20 COMMAND_CLASS_BASIC
+		 * 0x26 COMMAND_CLASS_SWITCH_MULTILEVEL
+		 * 0X27 COMMAND_CLASS_SWITCH_ALL
+		 * 0x33 COMMAND_CLASS_SWITCH_COLOR
+		 * 0x70 COMMAND_CLASS_CONFIGURATION
+		 * 0x73 COMMAND_CLASS_POWERLEVEL
+		 *
+		 * Here are the command classes used by this driver that we can fingerprint against:
+		 *
+		 * * 0x26 COMMAND_CLASS_SWITCH_MULTILEVEL -> yes, it is dimmable
+		 * * 0x33 COMMAND_CLASS_SWITCH_COLOR -> yes, it has color control
+		 */
+
+		// GENERIC_TYPE_SWITCH_MULTILEVEL:SPECIFIC_TYPE_POWER_SWITCH_MULTILEVEL
+		// dimmable, color control
+		fingerprint deviceId: "0x1101", inClusters: "0x26,0x33", deviceJoinName: "Z-Wave RGBW Bulb"
+
+		// GENERIC_TYPE_SWITCH_MULTILEVEL:SPECIFIC_TYPE_COLOR_TUNABLE_MULTILEVEL
+		// dimmable, color control
+		fingerprint deviceId: "0x1102", inClusters: "0x26,0x33", deviceJoinName: "Z-Wave RGBW Bulb"
```

Comment:
The Aeon Led Bulb DTH has very similar fingerprints. Is the thinking that this DTH replaces that DTH? If that's the case I think we should remove the fingerprints from the Aeon Led Bulb DTH.

Reply:
Yeah, @msandstedt If we can migrate over the Aeon LED Bulb fingerprint over to this DTH and remove that DTH. If there are any manufacturer specific changes such as setting configuration parameters, we can implement those in the DTH by checking the manufacturer/model of the device(Example: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/ed426552b4ffece81cb476a0d5614e7a300da888/devicetypes/smartthings/zwave-siren.src/zwave-siren.groovy#L466)

Reply:
@greens had a similar question in the Jira issue.  My thinking:

* yes, for sure we should migrate generic RGBW bulbs from the Aeon driver to this one; if they are like the Illumin, they likely won't work quite right in the Aeon DTH
* BUT, I am scared to unify in a single DTH because the Aeon-specific configuration commands are slightly complicated and cryptic to me
* I'm worried about introducing regressions for Aeon bulbs

Thoughts?

Reply:
OK, now I see that while aeon-led-bulb-6.groovy is Aeon-specific, aeon-led-bulb.groovy is attempting to very generically match to many dimmable bulbs with device type and inClusters.

I see aeon-led-bulb.groovy as somewhat broken and probably less good than the current driver.  So how should I handle this?

Reply:
Approach agreed upon offline:

* migrate all generic fingerprinting into existing rgbw-light.groovy
* replace body of rgbw-light.groovy with new code (new code works better)
* remove generic fingerprinting from aeon-led-bulb.groovy
* add mfr-specific fingerprint to aeon-led-bulb.groovy

Reply:
Note I did all of the above except this:

* add mfr-specific fingerprint to aeon-led-bulb.groovy

The mfr-specific Aeon fingerprint is already in rgbw-light.groovy; Aeon bulbs aren't actually matching to aeon-led-bulb.groovy in the first place, and Sumo confirms this.

Considering this resolved.